File: backend/src/app/schemas/aoi.py
from app.geo.schemas import GeoJSONFeature

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AOICreate(BaseModel):
    name: str
    description: str | None = None

    feature: GeoJSONFeature

class AOIUpdate(BaseModel):
    name: str | None = None
    description: str | None = None

    feature: GeoJSONFeature | None = None


class AOIResponse(BaseModel):
    id: int

    name: str
    description: str | None

    project_id: int

    created_at: datetime
    updated_at: datetime

    feature: GeoJSONFeature

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/src/types/aoi.ts
export interface GeoJSONPolygon {
  type: "Polygon";
  coordinates: number[][][];
}

export interface GeoJSONFeature {
  type: "Feature";
  geometry: GeoJSONPolygon;
  properties?: Record<string, unknown>;
}

export interface AOICreate {
  name: string;
  description?: string | null;
  feature: GeoJSONFeature;
}

export interface AOIUpdate {
  name?: string | null;
  description?: string | null;
  feature?: GeoJSONFeature | null;
}

export interface AOIResponse {
  id: number;
  name: string;
  description: string | null;
  project_id: number;
  created_at: string;
  updated_at: string;
  feature: GeoJSONFeature;
}
